### ISSUE TITLE
Private constructor of the ByteArrayBuilder doesn't use the BufferRecycler parameter

### ISSUE THREAD

**Issue by eneim** [FasterXML/jackson-core#1058]
This is only a question:

I notice the following [(private) constructor](https://github.com/FasterXML/jackson-core/blob/76d0749ff072acb7fdf4a924b80e7e347b192451/src/main/java/com/fasterxml/jackson/core/util/ByteArrayBuilder.java#L65C27-L65C27) of the `ByteArrayBuilder`:

```
private ByteArrayBuilder(BufferRecycler br, byte[] initialBlock, int initialLen) {
    _bufferRecycler = null;
    _currBlock = initialBlock;
    _currBlockPtr = initialLen;
}
```

I want to ask why `_bufferRecycler` is initialized to `null` given the constructor provides a `BufferRecycler`? Is that by intent or a mistake? If it is by intent, can you explain why?

**Comment by pjfanning** [comment]
@eneim thanks for raising this - in practise, the only code that uses this constructor does not have a bufferRecycler anyway but I created #1059 anyway

**Comment by cowtowncoder** [comment]
I'd venture a guess that this is accidental oversight. Thank you @eneim for reporting it.

**Comment by cowtowncoder** [comment]
Fixed via #1059
